### PR TITLE
Change unit testing process

### PR DESCRIPTION
**Pull request by Joonpark13**
After much thought, I've decided to remove our 100% test coverage requirement. I think this was actually providing less value than we thought. I also think that by removing this requirement, it will force us to pay more attention to which tests we are writing and why we should be writing them. This means everyone will (hopefully) better understand the unit tests we are writing.

### CODE REVIEW

**Comment by Joonpark13** [comment]
@kevinjihwanlee can you go through the process I've outline in the new doc when you get a chance and give me some feedback?

File: jest.config.js [hunk 1]
```diff
@@ -17,10 +17,10 @@ module.exports = {
   coverageReporters: process.env.COVERAGE_REPORT === 'on' ? ['html', 'text'] : ['html'],
   coverageThreshold: {
     global: {
-      branches: 100,
+      branches: 75,
       functions: 100,
-      lines: 100,
-      statements: 100,
+      lines: 75,
+      statements: 50,
```
Comment (Joonpark13) [inline jest.config.js:23]: We can circle back to see if these make sense down the line.

File: docs/UNIT_TESTING.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+First, let's fix our existing tests. Your `App.test.js` should now look like:
+
+```javascript
+import React from 'react';
+import { shallow } from 'enzyme';
+import App from './App';
+
+describe('App', () => {
+  it('renders without crashing', () => {
+    const wrapper = shalllow(<App />);
```
Comment (kevinjihwanlee) [inline docs/UNIT_TESTING.md]: small typo: should be `const wrapper = shallow(<App />);`

File: docs/UNIT_TESTING.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+But now, we don't assert that the button contains the text we expect. The action of switching the text isn't `MyComponent`'s job anymore - it's `App`'s. `MyComponent`'s job is to simply call the provided `onBtnClick` prop function when the user clicks the button. So, we make sure that this is what happened by asserting on the mock we provided as a prop:
+
+```javascript
+expect(onBtnClickMock).toHaveBeenCalled();
+```
+
+This is for the most part how you will test the interaction of child components with their parent - by providing props when setting up your test, and asserting on any mocks that you may have created.
+
+## Philosphy
```
Comment (kevinjihwanlee) [inline docs/UNIT_TESTING.md]: small typo: should be Philosophy

**Review by kevinjihwanlee — APPROVED** [review]
Everything makes sense to me, just pointed out a few small typos

**Pull request merged**